Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,373 @@
+        SentryTracer.resetAppStartMeasurementRead()
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+        SentryFramesTracker.sharedInstance().resetFrames()
+        SentryFramesTracker.sharedInstance().stop()
+#endif
+    }
+
+    // This test is only available on newer APIs because of the availability of `DispatchQueue.activate`
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testConcurrentProfilingTransactions() {
+        let options = fixture.options
+        options.profilesSampleRate = 1.0
+        options.tracesSampleRate = 1.0
+
+        let queue = DispatchQueue(label: "SentryProfilerSwiftTests", attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+
+        let numberOfTransactions = 10
+        for _ in 0 ..< numberOfTransactions {
+            group.enter()
+            let exp = expectation(description: "finished span")
+            let span = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+
+            // Some busy work to try and get it to show up in the profile.
+            let str = "a"
+            var concatStr = ""
+            for _ in 0..<100_000 {
+                concatStr = concatStr.appending(str)
+            }
+
+            queue.asyncAfter(deadline: .now() + 2) {
+                span.finish()
+                exp.fulfill()
+                group.leave()
+            }
+        }
+
+        queue.activate()
+        group.wait()
+        waitForExpectations(timeout: 5)
+
+        guard let envelope = self.fixture.client.captureEnvelopeInvocations.first else {
+            XCTFail("Expected to capture at least 1 event")
+            return
+        }
+        XCTAssertEqual(1, envelope.items.count)
+        guard let profileItem = envelope.items.first else {
+            XCTFail("Expected at least 1 additional envelope item")
+            return
+        }
+        XCTAssertEqual("profile", profileItem.header.type)
+        self.assertValidProfileData(data: profileItem.data, numberOfTransactions: numberOfTransactions)
+    }
+
+    func testConcurrentSpansWithTimeout() {
+        let options = fixture.options
+        options.profilesSampleRate = 1.0
+        options.tracesSampleRate = 1.0
+
+        let expA = expectation(description: "Span A finishes")
+        let spanA = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+
+        // Some busy work to try and get it to show up in the profile.
+        let str = "a"
+        var concatStr = ""
+        for _ in 0..<100_000 {
+            concatStr = concatStr.appending(str)
+        }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 45) {
+            spanA.finish()
+            expA.fulfill()
+        }
+
+        let expB = expectation(description: "Span B finishes")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 35) {
```

Comment:
`m`: This will slow down our test suite a lot. I would either make the timeout configurable, so we can reduce the timeout in tests, or we could use an abstraction for the NSTimer, which we substitute in tests to control when we want to execute the timeout manually. You can get some inspiration on how to write such tests here https://github.com/getsentry/sentry-cocoa/blob/22b13b71edc9ecbf67b3670ae7c4ba0e056bc9be/Tests/SentryTests/Performance/SentryTracerTests.swift#L180-L363

Reply:
I'll try the configurable timeout route with an internal-only API to do so.

Reply:
I wound up exposing the method to time out the profiler for usage in the tests in https://github.com/getsentry/sentry-cocoa/pull/2105/commits/f54851ae320999281a8ed6d7d39060b9e07affde, so we don't have to wait like this.